engine/collection/ingest/engine.go

```diff
@@ -65,13 +73,44 @@ func New(
 		},
 	)
 
+	// FIFO queue for transactions
+	queue, err := fifoqueue.NewFifoQueue(
+		fifoqueue.WithCapacity(int(config.MaxMessageQueueSize)),
+		fifoqueue.WithLengthObserver(func(len int) {
+			mempoolMetrics.MempoolEntries(metrics.ResourceTransactionIngestQueue, uint(len))
+		}),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("could not create transaction message queue: %w", err)
+	}
+	pendingTransactions := &engine.FifoMessageStore{FifoQueue: queue}
+
+	// define how inbound messages are mapped to message queues
+	handler := engine.NewMessageHandler(
+		logger,
+		engine.NewNotifier(),
+		engine.Pattern{
+			Match: func(msg *engine.Message) bool {
+				_, ok := msg.Payload.(*flow.TransactionBody)
+				if ok {
+					engMetrics.MessageReceived(metrics.EngineCollectionIngest, metrics.MessageTransaction)
+				}
+				return ok
+			},
+			Store: pendingTransactions,
+		},
+	)
+
 	e := &Engine{
-		unit:                 engine.NewUnit(),
+		lm:                   lifecycle.NewLifecycleManager(),
+		wg:                   new(sync.WaitGroup),
```

Comment:
why do we need a wait group? I don't see where we wait

Reply:
It's for the processing loop; https://github.com/onflow/flow-go/pull/2046/files#diff-09bd1de20e1e535512432a5e994d212b95f89937bda8a4df0020e9875db5e3bfR132-R133

Reply:
rather than using a `LifecycleManager` and `WaitGroup`, maybe this could use a `ComponentManager`. Right now this mixes the interfaces a bit and it may get confusing.

e.g. after engine is instantiated, add something like
```
	builder := component.NewComponentManagerBuilder().
		AddWorker(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
			ready()
			e.loop(ctx)
		})

	e.cm = builder.Build()
	e.Component = e.cm
```

then you can ditch the explicit ready/done/start methods since the `Component` interface is provided by `cm`

Reply:
@jordanschalm I think his point is that nowhere do we actually ever call `wg.Wait()`, so it's not actually be used.